src/transformers/modeling_utils.py

```diff
@@ -1569,6 +1587,65 @@ def retrieve_modules_from_names(self, names, add_prefix=False, remove_prefix=Fal
 
         return retrieved_modules
 
+    @classmethod
+    def _low_cpu_mem_usage_load(cls, model, loaded_state_dict_keys, resolved_archive_file):
+
+        # dematerialize param storage for keys that are going to be replaced by state_dict, by
+        # putting those on the meta device
+        for k in loaded_state_dict_keys:
+            split_name = k.split(".")
+            m = model
+            matched = False
+            while len(split_name) > 1:
+                if hasattr(m, split_name[0]):
+                    matched = True
+                    m = getattr(m, split_name[0])
+                    del split_name[0]
+                else:
+                    matched = False
+                    break
+            if matched:
+                # selectively switch to the meta device only those params/buffers that will
+                # be next replaced from state_dict. This a complex way to do p.to_("meta")
+                # since we have no in-place to_ for tensors.
+                to_meta_objs = []
+                for n, p in m.named_parameters():
+                    if n == split_name[0]:
+                        to_meta_objs.append(p.data_ptr())
+
+                def selective_to_meta(p):
+                    if p.data_ptr() in to_meta_objs:
```

Comment:
using `p` above and `is in` here should avoid the need to `data_ptr()` that might be problematic for some backends.

Reply:
Indeed! Thank you! Plus I had a suggestion from @cbalioglu, so the 2 possible solutions are:

```
            if matched:
                p = getattr(m, split_name[0])
                def selective_to_meta(t):
                    return t.to("meta") if t is p else t
                m._apply(selective_to_meta)
```
or:
```
            if matched:
                new_val = getattr(m, split_name[0])
                if isinstance(new_val, torch.nn.Parameter):
                    # isinstance returns False for Params on meta device
                    new_val = torch.nn.Parameter(new_val.to("meta"))
                else:
                    new_val = new_val.to("meta")
                setattr(m, split_name[0], new_val)
```

The latter seems to be easier to understand for users.